### ISSUE TITLE
Profiled Lens Correction inoperative in 5.8 (Windows)

### ISSUE THREAD

**Issue by Dave-McAllister** [Beep6581/RawTherapee#5633]
In v5.8 Windows 10 x64:

Automatic selection does not retrieve Camera/lens
Manual selection is unresponsive (no dropdown)

These function as expected in v5.7
Operating on the same image: Left 5.7, Right 5.8
![5758](https://user-images.githubusercontent.com/46267152/73836326-e6f45f80-4806-11ea-932d-5611155bf456.png)

Regards, David McAllister

**Comment by Benitoite** [comment]
Are the profiles present in the bundle?

**Comment by Dave-McAllister** [comment]
I don't know about the bundle as I just humbly run the installer but I've compared the installed folders/files and find that unlike 5.7, the Rawtherapee/5.8/share/lensfun folder is empty.

**Comment by Dave-McAllister** [comment]
I just tried copying the contents of 5.7...lensfun to 5.8...lensfun and Profiled Lens Correction now works. This workaround will do me for now but the underlying problem needs attention.
Thanks for the pointer.

**Comment by Benitoite** [comment]
@Dave-McAllister  Thanks for looking into it.  Seems like a basic packaging error.

**Comment by Beep6581** [comment]
The RawTherapee 5.8 installer for Windows is missing the lensfun DB. On it.

**Comment by aferrero2707** [comment]
I am having a look now.

**Comment by Beep6581** [comment]
@aferrero2707  I committed a fix (hopefully), waiting for it to recompile...
https://github.com/aferrero2707/rt-win64/issues/2

**Comment by aferrero2707** [comment]
Looks like the ARCH repositories now have LensFun v0.3.95, which ships a new version of the database:
```
[0;1mPackages (1)[0m lensfun-0.3.95-1

[0;1mTotal Download Size: [0m  0.34 MiB
[0;1mTotal Installed Size:[0m  3.01 MiB

[1;34m::[0;1m Proceed with installation? [Y/n] [0m
[1;34m::[0;1m Retrieving packages...
[0m lensfun-0.3.95-1...     0.0   B  0.00   B/s 00:00 [----------------------]   0%
 lensfun-0.3.95-1...    48.0 KiB  72.7 KiB/s 00:04 [##--------------------]  13%
 lensfun-0.3.95-1...   224.0 KiB   315 KiB/s 00:00 [#############---------]  63%
 lensfun-0.3.95-1...   351.4 KiB   799 KiB/s 00:00 [######################] 100%
(0/1) checking keys in keyring                     [----------------------]   0%
(1/1) checking keys in keyring                     [######################] 100%
(0/1) checking package integrity                   [----------------------]   0%
(1/1) checking package integrity                   [######################] 100%
(0/1) loading package files                        [----------------------]   0%
(1/1) loading package files                        [######################] 100%
(0/1) checking for file conflicts                  [----------------------]   0%
(1/1) checking for file conflicts                  [######################] 100%
(0/1) checking available disk space                [----------------------]   0%
(1/1) checking available disk space                [######################] 100%
[1;34m::[0;1m Processing package changes...
[0m(1/1) installing lensfun                           [----------------------]   0%
(1/1) installing lensfun                           [######################] 100%
Optional dependencies for lensfun
[0;1m   [0m python: for lensfun-update-data and lensfun-add-adapter [installed]
[1;34m::[0;1m Running post-transaction hooks...
[0m(1/1) Arming ConditionNeedsUpdate...
sudo: setrlimit(RLIMIT_CORE): Operation not permitted
Reading http://lensfun.sourceforge.net/db/versions.json â€¦
Reading http://wilson.bronger.org/lensfun-db/versions.json â€¦
Successfully updated the database in /var/lib/lensfun-updates/version_2.
```
The mingw version that is used to compile RT is v0.3.2, which I suspect still requires a version_1 DB.

I am modifying the scripts to use `lensfun-update-data` from 0.3.2 instead of 0.3.95.

**Comment by Beep6581** [comment]
@aferrero2707 ok, I will leave this to you.

**Comment by aferrero2707** [comment]
The builds are in progress. Sorry for overlooking this detail...

**Comment by aferrero2707** [comment]
@Beep6581 @heckflosse @Dave-McAllister the bundled LensFun DB should be OK again in the windows packages. Could somebody please confirm?

I did a quick test under wine (no, I'm not getting drunk, don't worry....) and the camera/lens combo is now automatically detected in my case:

![Screen Shot 2020-02-06 at 9 18 08 AM](https://user-images.githubusercontent.com/6229237/73918405-dead2a80-48c1-11ea-8a0f-d0223cf3fbf4.png)

**Comment by Beep6581** [comment]
Get the latest build here:
https://github.com/Beep6581/RawTherapee/releases/download/nightly/RawTherapee_5.8.exe
sha256sum:
`2d6f127598ef9a3d36a86657ac53c81b1957b6d9ae97fef1fadd8757f82aa915  RawTherapee_5.8.exe`

I confirm that the latest build includes the Lensfun DB, tested through wine.

@aferrero2707 I get some missing icons and a corrupt main status bar label - is this a wine-only problem? Tested using wine-4.16.

![screenshot_20200206_093053](https://user-images.githubusercontent.com/5844619/73919412-cc33f080-48c3-11ea-8ffe-e54621da3ffd.png)

I will upload the build to the website once a user of real Windows confirms that it works correctly - lensfun DB present and no missing icons.

**Comment by Thanatomanic** [comment]
On it!

**Comment by Thanatomanic** [comment]
I can confirm: lensfun works as intended, no icons are missing.

May I suggest to release this as 5.8.1, to make clear it contains an important fix?

**Comment by Beep6581** [comment]
@Thanatomanic RT's code hasn't changed at all, so I'll just do a switcheroo and write a news post.

**Comment by Beep6581** [comment]
Will wait for [the macOS build](https://discuss.pixls.us/t/rawtherapee-macos-build-needs-urgent-testing/16176/26?u=morgan_hardwood) from @Benitoite before uploading the new Windows build, and announce this in one news post.

**Comment by arturoisilvia** [comment]
Tried:
https://github.com/Beep6581/RawTherapee/releases/tag/nightly/RawTherapee_5.8.exe
Lensfun now works correctly on my windows 10 & RT-5.8
I only confirm that I have installed it and it works fine. 
(although I am more of the LocalLab branch....)
Greetings.

**Comment by Benitoite** [comment]
@Beep6581 @aferrero2707 Just before the 5.8 tag we had 2cc8918 which removed
https://github.com/Beep6581/RawTherapee/blob/364fe35023874276b9572f58a62b9e62787bc981/CMakeLists.txt#L255-L257
#5639 reverts and adds `if (NOT APPLE)`

**Comment by Beep6581** [comment]
PR #5639 merged in commit 113f096 (ping @rfranke)

Ok to close?

**Comment by rfranke** [comment]
Sure if other systems require this.
